### ISSUE TITLE
Transaction support?

### ISSUE THREAD

**Issue by dstroot** [gchaincl/dotsql#10]
I need to update several tables at once and would like that to be in one transaction.  Can't seem to figure out how dotsql and transactions can play together nicely and noticed you have no examples either.  Thoughts?

**Comment by gchaincl** [comment]
Not sure if transactions should concern dotsql. You can start a transaction with (https://golang.org/pkg/database/sql/#DB.Begin) and the call queries loaded from dotsql.

**Comment by gchaincl** [comment]
closing this, but if you think I'm missing something, feel free to reopen it

**Comment by erthalion** [comment]
> Not sure if transactions should concern dotsql. You can start a transaction with [begin](https://golang.org/pkg/database/sql/#DB.Begin) and the call queries loaded from dotsql.

Maybe I'm missing something, but in this case you need to get a raw query and execute it manually, which is probably less convenient, that just have a wrapper around Tx in the library itself.

**Comment by gchaincl** [comment]
@erthalion how would you do it? Could you provide pseudo-code to better understand your idea?

**Comment by erthalion** [comment]
For now I'm doing something like that in my code, is that what you're asking?:

```
func (d DotSql) ExecTx(tx *sql.Tx, name string, args ...interface{}) (sql.Result, error) {
	query, err := d.lookupQuery(name)
	if err != nil {
		return nil, err
	}

	return tx.Exec(query, args...)
}
```

**Comment by gchaincl** [comment]
@erthalion I understand better now, it shouldn't be hard to implement right?

**Comment by erthalion** [comment]
Yeah, of course - just originally I had strong feeling that such simple thing is already implemented here somewhere :)

**Comment by gchaincl** [comment]
Feel free to open a PR, I'll be happy to review it

**Comment by AJBrindley** [comment]
Why was ExecTx removed form the v1.0.0 release?  I was utilising this.

**Comment by gchaincl** [comment]
Which version / commit hash are you using?
